plugins/CoreConsole/Commands/DevelopmentSyncUITestScreenshots.php

```diff
@@ -73,42 +71,166 @@ protected function execute(InputInterface $input, OutputInterface $output)
             if ($file !== null
                 && preg_match("/" . $screenshotsRegex . "/", $file)
             ) {
-                if ($testPlugin == null) {
-                    $downloadTo = "tests/UI/expected-ui-screenshots/$file";
-                } else {
-                    $downloadTo = "plugins/$testPlugin/tests/UI/expected-ui-screenshots/$file";
-                }
-
-                $output->write("<info>Downloading $file to  $downloadTo...</info>\n");
-                Http::sendHttpRequest("$urlBase/processed-ui-screenshots/$file", $timeout = 60, $userAgent = null,
-                    PIWIK_DOCUMENT_ROOT . "/" . $downloadTo);
+                $this->downloadProcessedScreenshot($input, $output, $file);
             }
         }
 
-        $this->displayGitInstructions($output);
+        $this->displayGitInstructions($input, $output);
 
     }
 
-    /**
-     * @param OutputInterface $output
-     */
-    protected function displayGitInstructions(OutputInterface $output)
+    protected function displayGitInstructions(InputInterface $input, OutputInterface $output)
     {
         $output->writeln('');
         $output->writeln('--------------');
         $output->writeln('');
         $output->writeln("If all downloaded screenshots are valid you may push them with these commands:");
-        $output->writeln('');
-        $commands = "cd tests/UI/expected-ui-screenshots
+        $downloadToPath = $this->getDownloadToPath($input);
+        $commands = "
+cd $downloadToPath
 git pull
 git add .
-git commit -m '' # WRITE A COMMIT MESSAGE
-git push
+git commit -m '' # Write a good commit message
+git push";
+
+        $plugin = $input->getOption('plugin');
+        if(empty($plugin)) {
+            $commands .= "
 cd ..
 git pull
-git add expected-ui-screenshots
-git commit -m '' #WRITE A COMMIT MESSAGE
-git push";
+git add expected-ui-screenshots/
+git commit -m '' # Copy paste the good commit message
+git push
+cd ../../\n\n";
+        } else {
+            $commands .= "
+cd ../../../../../\n\n";
+        }
         $output->writeln($commands);
     }
+
+    protected function getUrlBase(InputInterface $input)
+    {
+        $buildNumber = $input->getArgument('buildnumber');
+        if (empty($buildNumber)) {
+            throw new \InvalidArgumentException('Missing build number.');
+        }
+
+        $plugin = $input->getOption('plugin');
+        if ($plugin) {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master.%s/%s', $plugin, $buildNumber);
+        } else {
+            $urlBase = sprintf('http://builds-artifacts.piwik.org/ui-tests.master/%s', $buildNumber);
+        }
+        return $urlBase;
```

Comment:
It might increase clarity to move the sprintf's to private methods w/ descriptive names. I'm not sure though, so maybe it won't.